src/components/tickets/index.tsx

```diff
@@ -0,0 +1,104 @@
+import CustomToolbar from './customToolbar';
+import { ITicketState } from '../../types/tickets';
+import MenuCell from './cells/menuCell';
+import TicketsModal from '../modals/tickets';
+import { deleteTicketAsync } from '../../store/tickets/thunk';
+import { useAppDispatch } from '../../store/hooks';
+import { useSelector } from 'react-redux';
+import { useTableSortAndFilter } from './useTableSort';
+import { StyledBox, StyledDataGrid } from './styled';
+import { baseMenuCellConfig, columns, pageSizeOptions } from './helper';
+import { tickets, totalRows } from '../../store/tickets/selector';
+import { useCallback, useEffect, useState } from 'react';
+
+const TicketsTable = () => {
+  const dispatch = useAppDispatch();
+
+  const {
+    searchParams,
+    fetchTickets,
+    setSelectedPriorities,
+    selectedPriorities,
+    paginationModel,
+    setPaginationModel,
+  } = useTableSortAndFilter();
+
+  useEffect(() => {
```

Comment:
Can we move useEffect to the `useTableSortAndFilter`?

Reply:
fix: useEffect moved to useTableSortAndFilter